crates/matrix-sdk-sqlite/src/state_store.rs

```diff
@@ -0,0 +1,1426 @@
+        Ok(())
+    }
+
+    fn set_display_name(&self, room_id: &[u8], name: &[u8], data: &[u8]) -> rusqlite::Result<()> {
+        self.prepare_cached(
+            "INSERT OR REPLACE
+             INTO display_name (room_id, name, data)
+             VALUES (?1, ?2, ?3)",
+        )?
+        .execute((room_id, name, data))?;
+        Ok(())
+    }
+
+    fn remove_display_name(&self, room_id: &[u8], name: &[u8]) -> rusqlite::Result<()> {
+        self.prepare("DELETE FROM display_name WHERE room_id = ? AND name = ?")?
+            .execute((room_id, name))?;
+        Ok(())
+    }
+
+    fn remove_room_display_names(&self, room_id: &[u8]) -> rusqlite::Result<()> {
+        self.prepare("DELETE FROM display_name WHERE room_id = ?")?.execute((room_id,))?;
+        Ok(())
+    }
+}
+
+#[async_trait]
+trait SqliteObjectStateStoreExt: SqliteObjectExt {
+    async fn get_kv_blob(&self, key: Key) -> Result<Option<Vec<u8>>> {
+        Ok(self
+            .query_row("SELECT value FROM kv_blob WHERE key = ?", (key,), |row| row.get(0))
+            .await
+            .optional()?)
+    }
+
+    async fn set_kv_blob(&self, key: Key, value: Vec<u8>) -> Result<()>;
+
+    async fn delete_kv_blob(&self, key: Key) -> Result<()> {
+        self.execute("DELETE FROM kv_blob WHERE key = ?", (key,)).await?;
+        Ok(())
+    }
+
+    async fn get_room_infos(&self, stripped: bool) -> Result<Vec<Vec<u8>>> {
+        Ok(self
+            .prepare("SELECT data FROM room_info WHERE stripped = ?", move |mut stmt| {
+                stmt.query_map((stripped,), |row| row.get(0))?.collect()
+            })
+            .await?)
+    }
+
+    async fn get_state_event(
+        &self,
+        room_id: Key,
+        event_type: Key,
+        state_key: Key,
+        stripped: bool,
+    ) -> Result<Option<Vec<u8>>> {
+        Ok(self
+            .query_row(
+                "SELECT data FROM state_event
+                 WHERE room_id = ? AND event_type = ? AND state_key = ? AND stripped = ?",
+                (room_id, event_type, state_key, stripped),
+                |row| row.get(0),
+            )
+            .await
+            .optional()?)
+    }
+
+    async fn get_maybe_stripped_state_event(
+        &self,
+        room_id: Key,
+        event_type: Key,
+        state_key: Key,
+    ) -> Result<Option<(bool, Vec<u8>)>> {
+        Ok(self
+            .query_row(
+                "SELECT stripped, data FROM state_event
+                 WHERE room_id = ? AND event_type = ? AND state_key = ?",
+                (room_id, event_type, state_key),
+                |row| Ok((row.get(0)?, row.get(1)?)),
+            )
+            .await
+            .optional()?)
+    }
+
+    async fn get_state_events(&self, room_id: Key, event_type: Key) -> Result<Vec<Vec<u8>>> {
+        Ok(self
+            .prepare(
+                "SELECT data FROM state_event WHERE room_id = ? AND event_type = ? AND stripped = 0",
```

Comment:
```suggestion
                "SELECT data FROM state_event WHERE room_id = ? AND event_type = ? AND stripped = FALSE",
```

just for readability